runtime/include/cloe/data_broker.hpp

```diff
@@ -0,0 +1,2096 @@
+    if (!inserted.second) {
+      throw std::out_of_range(fmt::format("cannot alias signal '{}' to '{}': name already exists",
+                                          signal->name_or("<unnamed>"), new_name));
+    }
+    // signals mutated, there is a liability in case of exceptions
+    try {
+      SignalPtr result = inserted.first->second;
+      result->add_name(new_name);
+      return result;
+    } catch (...) {
+      // fullfill exception guarantee (map.erase(iter) does not throw)
+      signals_.erase(inserted.first);
+      throw;
+    }
+  }
+
+  /**
+   * Give an existing signal a (new) name.
+   *
+   * \param old_name Name of the existing signal
+   * \param new_name New name of the signal
+   * \param f flag_type flags used to guide the interpretation of the character sequence as a regular expression
+   * \return Pointer to the signal
+   * \note If an exception is thrown by any operation, the aliasing has no effect.
+   */
+  SignalPtr alias(std::string_view old_name, std::string_view new_name,
+                  std::regex::flag_type f = std::regex_constants::ECMAScript) {
+    std::regex regex = std::regex(std::string(old_name), f);
+    auto it1 = signals().begin();
+    auto it2 = signals().begin();
+    auto end = signals().end();
+    const auto predicate = [&](const auto& item) -> bool {
+      std::smatch match;
+      return std::regex_match(item.first, match, regex);
+    };
+    it1 = (it1 != end) ? std::find_if(it1, end, predicate) : end;
+    it2 = (it1 != end) ? std::find_if(std::next(it1), end, predicate) : end;
+    if (it2 != end) {
+      throw std::out_of_range(
+          fmt::format("regex pattern matches multiple signals: '{}'; matches: '{}', '{}'", old_name,
+                      it1->first, it2->first));
+    }
+    if (it1 == end) {
+      throw std::out_of_range(fmt::format("regex pattern matches zero signals: {}", old_name));
+      return nullptr;
+    }
+    SignalPtr result = alias(it1->second, new_name);
+    return result;
+  }
+
+  /**
+   * Declare a new signal.
+   *
+   * \tparam T Type of the signal
+   * \param name Name of the signal
+   * \return Pointer to the specified signal
+   */
+  template <typename T>
+  SignalPtr declare(std::string_view name) {
+    assert_static_type<T>();
+    using compatible_type = databroker::compatible_base_t<T>;
+
+    declare<compatible_type>();
+
+    SignalPtr signal = Signal::make<compatible_type>();
+    alias(signal, name);
+    return signal;
+  }
+
+  /**
+   * Declare a new signal and auto-implement getter and setter.
+   *
+   * \tparam T type of the signal value
+   * \param name name of the signal
+   * \param value_ptr pointer to signal value
+   * \return pointer to specified signal
+   */
+  template <typename T>
+  SignalPtr declare(std::string_view name, T* value_ptr) {
+    assert(value_ptr != nullptr);
+    auto signal = this->declare<T>(name);
+    signal->template set_getter<T>([value_ptr]() -> const T& {
```

Comment:
Does this compile? Maybe its me, but is 

signal->template set_getter<T>(...);

proper C++ syntax?